internal/bminventory/inventory.go

```diff
@@ -751,6 +751,10 @@ func (b *bareMetalInventory) createAndUploadDay2NodeIgnition(ctx context.Context
 		caCert = cluster.IgnitionEndpoint.CaCertificate
 	}
 
+	if host.IgnitionConfigOverrides != "" {
+		ignitionEndpointUrl = fmt.Sprintf("https://%s:22623/config/%s", address, host.MachineConfigPoolName)
```

Comment:
When cluster is being created in ZTP each host gets special annotation with MCS certificate - see https://github.com/openshift/assisted-service/blob/1cfbfc7ece4cd6d6bf139347650afd552d30d770/internal/controller/controllers/bmh_agent_controller.go#L1150-L1155

Reply:
I know but host can have this variable set by user in saas env and we don't want to move to https in that case as we don't even know that user pushed certificate. 
I think all we need is to change cluster.IgnitionEndpoint var if ensureMCSCert succeeded.

Reply:
This can be done in controller and will not require adding logic to common code

Reply:
So, a better option would be `ensureMCSCert` setting a certain flag / Ignition URL?

Reply:
somewhere in that area we can set IgnitionEndpoint , @eranco74 what do you think?

Reply:
Right, I guess instead of this change we could do:
* set custom Ignition URL in `ensureMCSCert` to 22623
* follow the tested "custom Ignition URL is set" flow in createAndUploadDay2NodeIgnition